flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -503,15 +504,37 @@ class TaskManager(
             )
           }
 
-        case PartitionState(taskExecutionId, taskResultId, partitionId, state) =>
-          Option(runningTasks.get(taskExecutionId)) match {
+        // Updates the partition producer state
+        case PartitionProducerState(receiverExecutionId, result) =>
+          Option(runningTasks.get(receiverExecutionId)) match {
             case Some(task) =>
-              task.onPartitionStateUpdate(taskResultId, partitionId, state)
+              try {
+                result match {
+                  case Left((intermediateDataSetId, resultPartitionId, producerState)) =>
+                    // Forward the state update to the task
+                    task.onPartitionStateUpdate(
+                      intermediateDataSetId,
+                      resultPartitionId.getPartitionId,
+                      producerState)
+
+                  case Right(failure) =>
+                    // Cancel or fail the execution
+                    if (failure.isInstanceOf[PartitionProducerDisposedException]) {
+                      log.debug("Partition producer disposed. Cancelling execution.", failure)
```

Comment:
I think this log statement should be on `info` level. Otherwise, going through the logs in standard settings (only info level logging) leaves you wondering why the task is cancelled all of a sudden.